specs/Standards/Wallets/InjectedWallets.md

```diff
@@ -0,0 +1,369 @@
+
+For dApps that often sign gas-only transactions, [`FunctionCall`](https://nomicon.io/DataStructures/AccessKey#accesskeypermissionfunctioncall) access keys can be added/deleted for one or more accounts using the [`signIn`](#signin) and [`signOut`](#signout) methods. While this functionality could be achieved with [`signTransactions`](#signtransactions), it suggests a direct intention that a user wishes to sign in/out of a dApp's smart contract.
+
+### Wallet API
+
+Below is the entire API for injected wallets. It makes use of `near-api-js` to enable interoperability with dApps that will already use it for constructing transactions and communicating with RPC endpoints.
+
+```ts
+import { transactions, utils } from "near-api-js";
+
+interface Account {
+  accountId: string;
+  publicKey: utils.PublicKey;
+}
+
+interface Network {
+  networkId: string;
+  nodeUrl: string;
+}
+
+interface SignInParams {
+  permission: transactions.FunctionCallPermission;
+  accounts: Array<Account>;
+}
+
+interface SignOutParams {
+  accounts: Array<Account>;
+}
+
+interface SignTransactionParams {
+  transaction: transactions.Transaction;
+}
+
+interface SignTransactionsParams {
+  transactions: Array<transactions.Transaction>;
+}
+
+interface Events {
+  accountsChanged: { accounts: Array<Account> };
+  networkChanged: { network: Network };
+}
+
+type Unsubscribe = () => void;
+
+interface Wallet {
+  id: string;
+  connected: boolean;
+  network: Network;
+  accounts: Array<Account>;
+  
+  connect(): Promise<Array<Account>>;
+  signIn(params: SignInParams): Promise<void>;
+  signOut(params: SignOutParams): Promise<void>;
+  signTransaction(params: SignTransactionParams): Promise<transactions.SignedTransaction>;
+  signTransactions(params: SignTransactionsParams): Promise<Array<transactions.SignedTransaction>>;
+  disconnect(): Promise<void>;
+  on<EventName extends keyof Events>(
+    event: EventName,
+    callback: (params: Events[EventName]) => void
+  ): Unsubscribe;
+  off<EventName extends keyof Events>(
+    event: EventName,
+    callback?: () => void
+  ): void;
+}
+```
+
+#### Properties
+
+##### `id`
+
+Retrieve the wallet's unique identifier.
+
+```ts
+const { id } = window.near.wallet;
+
+console.log(id) // "wallet"
+```
+
+##### `connected`
+
+Determine whether we're already connected to the wallet and have visibility of at least one account.
+
+```ts
+const { connected } = window.near.wallet;
+
+console.log(connected) // true
+```
+
+##### `network`
+
+Retrieve the currently selected network.
+
+```ts
+const { network } = window.near.wallet;
+
+console.log(network) // { networkId: "testnet", nodeUrl: "https://rpc.testnet.near.org" }
+```
+
+##### `accounts`
+
+Retrieve all accounts visible to the dApp.
+
+```ts
+const { accounts } = window.near.wallet;
+
+console.log(accounts) // [{ accountId: "test.testnet", publicKey: PublicKey }]
+```
+
+#### Methods
+
+##### `connect`
```

Comment:
What should happen when another method is called before the first invocation of `connect`?

Reply:
Since the spec requires an account ID be provided for other actions such as `signIn` and `signTransaction`, the dApp / consumer of this API would typically not know which account ID to use for the payloads for those methods unless they have called `connect()` previously to identify which account IDs are supported by the currently selected  wallet.

We do not explicitly _require_ that `connect` is _always_ called before the other methods to facilitate cases where the dApp might use cached data from a prior wallet connection call, or data from another external source to compose payloads for the other methods.